srcjs/utils.js

```diff
@@ -246,3 +246,230 @@ function mapValues(obj, f) {
   }
   return newObj;
 }
+
+// Binary equality function used by the equal function.
+function _equal(x, y) {
+  if ($.type(x) === "object" && $.type(y) === "object") {
```

Comment:
Am I correct that you're using `$.type()` because there's not a good plain-JS way of checking if an object is a regular object (and not, for example, a Date object)?